--- v100/src/app/play/page.tsx	2025-08-27 01:15:12.000000000 +0800
+++ open-with-mpv/src/app/play/page.tsx	2025-09-16 14:28:31.717495826 +0800
@@ -149,6 +149,46 @@
   // 视频播放地址
   const [videoUrl, setVideoUrl] = useState('');
 
+  // safe base64 for utf8
+  const safeU8Base64 = (str: string) => {
+    const b64 = btoa(unescape(encodeURIComponent(str)));
+    return b64.replace(/\//g, "_").replace(/\+/g, "-").replaceAll("=", "");
+  };
+  // 根据 videoUrl 生成 mpv:// 播放地址
+  const genVideoMPVUrl = (
+    theVideoUrl: string,
+    theVideoTitle: string,
+    theDetailData: SearchResult | null,
+    theEpisodeIndex: number
+  ) => {
+    if (theVideoUrl) {
+      console.log('[MPV] 转换视频地址：', theVideoUrl);
+      const safeurl = safeU8Base64(theVideoUrl);
+      console.log('[MPV] 转换结果：', safeurl);
+      const v_title = (theDetailData?.episodes?.length || 0) > 1
+        ? `${theVideoTitle || '影片标题'} - ${theDetailData?.episodes_titles?.[theEpisodeIndex] || `第${theEpisodeIndex + 1}集`}`
+        : `${theVideoTitle || '影片标题'}`;
+      console.log('[MPV] 转换视频标题：', v_title);
+      const safetitle = safeU8Base64(v_title);
+      console.log('[MPV] 转换结果：', safetitle);
+      const mpvurl = `mpv://play/${safeurl}/?v_title=${safetitle}`;
+      console.log('[MPV] 播放地址：', mpvurl);
+      return mpvurl;
+    } else {
+      return '';
+    }
+  };
+  // 仅非移动设备，当 videoUrl 变化时，自动更新 mpv:// 播放地址
+  const [videoMPVUrl, setVideoMPVUrl] = useState('');
+  useEffect(() => {
+    const isMobile = typeof navigator !== 'undefined'
+        ? /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent)
+        : false;
+    if (!isMobile) {
+      setVideoMPVUrl(genVideoMPVUrl(videoUrl, videoTitle, detail, currentEpisodeIndex));
+    }
+  }, [videoUrl]);
+
   // 总集数
   const totalEpisodes = detail?.episodes?.length || 0;
 
@@ -1838,6 +1878,37 @@
                 {` > ${detail?.episodes_titles?.[currentEpisodeIndex] || `第 ${currentEpisodeIndex + 1} 集`}`}
               </span>
             )}
+            {/* 非移动设备，同一行 MPV button */}
+            {videoMPVUrl && videoMPVUrl.length > 0 && (
+            <button
+              onClick={() => {
+                const currentPlayTime = artPlayerRef.current?.currentTime || 0;
+                const currentvideoMPVUrl = currentPlayTime > 5
+                    ?`${videoMPVUrl}&startat=${Math.floor(currentPlayTime) - 1}`
+                    :videoMPVUrl;
+                // 暂停播放器
+                if (artPlayerRef.current && !artPlayerRef.current.paused) {
+                    artPlayerRef.current.pause();
+                }
+                // target name
+                //  "_blank": 新窗口。artPlayer 仅暂停。
+                //  "_self": 页面卸载时, handleBeforeUnload, artPlayer 资源会被清理，比仅暂停更省资源。(NOT-TODO) 可添加状态判断，阻止清理。
+                //  "open-with-mpv": specified 新窗口，避免多个 blank 窗口。
+                const targetName = "open-with-mpv";
+                const targetTitle = `${document.title} | MPV`;
+                const mpvindow =window.open(currentvideoMPVUrl, targetName);
+                if (mpvindow && mpvindow.document) {
+                    mpvindow.document.title = targetTitle;
+                }
+              }}
+              className='ml-4 flex-shrink-0 hover:opacity-90 transition-all duration-200 hover:scale-105'
+              title={videoUrl}
+            >
+              <div className='flex items-center gap-1.5 bg-blue-200 hover:bg-blue-400 dark:bg-blue-600 dark:hover:bg-blue-800 text-gray-600 dark:text-gray-300 px-3 py-1.5 rounded-full text-sm font-semibold shadow-md'>
+                <span>MPV 播放</span>
+              </div>
+            </button>
+            )}
           </h1>
         </div>
         {/* 第二行：播放器和选集 */}
