From c65ce8c55d61696b56bdd906ddb290440244e3c1 Mon Sep 17 00:00:00 2001
From: shinya <kobayashi_shinya@outlook.com>
Date: Thu, 25 Sep 2025 00:54:27 +0800
Subject: [PATCH] fix bangumi

---
 src/lib/bangumi.client.ts          |  7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/src/lib/bangumi.client.ts b/src/lib/bangumi.client.ts
index 678ebaeb6..8f569537a 100644
--- a/src/lib/bangumi.client.ts
+++ b/src/lib/bangumi.client.ts
@@ -25,5 +25,10 @@ export interface BangumiCalendarData {
 export async function GetBangumiCalendarData(): Promise<BangumiCalendarData[]> {
   const response = await fetch('https://api.bgm.tv/calendar');
   const data = await response.json();
-  return data;
+  const filteredData = data.map((item: BangumiCalendarData) => ({
+    ...item,
+    items: item.items.filter(bangumiItem => bangumiItem.images)
+  }));
+
+  return filteredData;
 }
