--- v100/src/app/play/page.tsx	2025-08-27 01:15:12.000000000 +0800
+++ open-with-mpv/src/app/play/page.tsx	2025-09-13 14:00:52.911743898 +0800
@@ -148,6 +148,7 @@
 
   // 视频播放地址
   const [videoUrl, setVideoUrl] = useState('');
+  const [videoMPVUrl, setVideoMPVUrl] = useState('');
 
   // 总集数
   const totalEpisodes = detail?.episodes?.length || 0;
@@ -407,6 +408,11 @@
     return Math.round(score * 100) / 100; // 保留两位小数
   };
 
+  // base64 utf8
+  const btou8 = (str: string) => {
+    return btoa(unescape(encodeURIComponent(str)));
+  };
+
   // 更新视频地址
   const updateVideoUrl = (
     detailData: SearchResult | null,
@@ -418,11 +424,23 @@
       episodeIndex >= detailData.episodes.length
     ) {
       setVideoUrl('');
+      setVideoMPVUrl('');
       return;
     }
     const newUrl = detailData?.episodes[episodeIndex] || '';
     if (newUrl !== videoUrl) {
+      console.log('更新视频地址为：', newUrl);
       setVideoUrl(newUrl);
+      const safeBase64Url = btou8(newUrl).replace(/\//g, "_").replace(/\+/g, "-").replaceAll("=", "");
+      const episodetitle = detailData.episodes.length > 1
+        ?`${videoTitle || '影片标题'} - ${detailData?.episodes_titles?.[episodeIndex] || `第${episodeIndex + 1}集`}`
+        :`${videoTitle || '影片标题'}`;
+      console.log('更新MPV-episodetitle为：', episodetitle);
+      const safeBase64title = btou8(episodetitle).replace(/\//g, "_").replace(/\+/g, "-").replaceAll("=", "");
+      console.log('更新MPV-encodetitle为：', safeBase64title);
+      const newMPVUrl = `mpv://play/${safeBase64Url}/?v_title=${safeBase64title}`;
+      console.log('更新MPV播放地址为：', newMPVUrl);
+      setVideoMPVUrl(newMPVUrl);
     }
   };
 
@@ -1838,6 +1856,37 @@
                 {` > ${detail?.episodes_titles?.[currentEpisodeIndex] || `第 ${currentEpisodeIndex + 1} 集`}`}
               </span>
             )}
+            {/* 同一行 MPV button */}
+            {videoUrl && videoUrl.length > 0 && (
+            <button
+              onClick={() => {
+                const currentPlayTime = artPlayerRef.current?.currentTime || 0;
+                const currentvideoMPVUrl = currentPlayTime > 5
+                    ?`${videoMPVUrl}&startat=${currentPlayTime}`
+                    :videoMPVUrl;
+                // 暂停播放器
+                if (artPlayerRef.current && !artPlayerRef.current.paused) {
+                    artPlayerRef.current.pause();
+                }
+                // target name
+                //  "_blank": 新窗口。artPlayer 仅暂停。
+                //  "_self": 页面卸载时, handleBeforeUnload, artPlayer 资源会被清理，比仅暂停更省资源。(NOT-TODO) 可添加状态判断，阻止清理。
+                //  "open-with-mpv": specified 新窗口，避免多个 blank 窗口。
+                const targetName = "open-with-mpv";
+                const targetTitle = `${document.title} | MPV`;
+                const mpvindow =window.open(currentvideoMPVUrl, targetName);
+                if (mpvindow && mpvindow.document) {
+                    mpvindow.document.title = targetTitle;
+                }
+              }}
+              className='ml-4 flex-shrink-0 hover:opacity-90 transition-all duration-200 hover:scale-105'
+              title={videoUrl}
+            >
+              <div className='flex items-center gap-1.5 bg-blue-200 hover:bg-blue-400 dark:bg-blue-600 dark:hover:bg-blue-800 text-gray-600 dark:text-gray-300 px-3 py-1.5 rounded-full text-sm font-semibold shadow-md'>
+                <span>MPV 播放</span>
+              </div>
+            </button>
+            )}
           </h1>
         </div>
         {/* 第二行：播放器和选集 */}
