diff --git a/src/lib/live.ts b/src/lib/live.ts
--- a/src/lib/live.ts
+++ b/src/lib/live.ts
@@ -1,5 +1,6 @@
 import { getConfig } from '@/lib/config';
 import { db } from '@/lib/db';
+import { completeCollectionPath } from '@/lib/config.collections';
 
 const defaultUA = 'AptvPlayer/1.4.10';
 
@@ -62,7 +63,7 @@ export async function refreshLiveChannels(liveInfo: {
     delete cachedLiveChannels[liveInfo.key];
   }
   const ua = liveInfo.ua || defaultUA;
-  const response = await fetch(liveInfo.url, {
+  const response = await fetch(completeCollectionPath(liveInfo.url, null), {
     headers: {
       'User-Agent': ua,
     },
