diff --git a/src/lib/live.ts b/src/lib/live.ts
index fa4a5e9..ff675e9 100644
--- a/src/lib/live.ts
+++ b/src/lib/live.ts
@@ -2,6 +2,7 @@
 
 import { getConfig } from "@/lib/config";
 import { db } from "@/lib/db";
+import { completeCollectionPath } from '@/lib/config.collections';
 
 const defaultUA = 'AptvPlayer/1.4.10'
 
@@ -62,7 +63,7 @@ export async function refreshLiveChannels(liveInfo: {
     delete cachedLiveChannels[liveInfo.key];
   }
   const ua = liveInfo.ua || defaultUA;
-  const response = await fetch(liveInfo.url, {
+  const response = await fetch(completeCollectionPath(liveInfo.url, null), {
     headers: {
       'User-Agent': ua,
     },
@@ -95,7 +96,7 @@ async function parseEpg(epgUrl: string, ua: string, tvgIds: string[]): Promise<{
   const result: { [key: string]: { start: string; end: string; title: string }[] } = {};
 
   try {
-    const response = await fetch(epgUrl, {
+    const response = await fetch(completeCollectionPath(epgUrl, null), {
       headers: {
         'User-Agent': ua,
       },
